### PR TITLE
[MH-71] Info about uploaded AD & delete view

### PR DESCRIPTION
**Pull request by nmashton**
Per requirements, this fills out the "current AD" widget option and adds a delete endpoint (as well as a test for that endpoint).

To make this work, it was necessary to somewhat alter the logic in the preceding steps to ensure that an AD actually got uploaded in the correct manner. Hopefully the way in which this was done is clear overall.

### CODE REVIEW

File: myhpom/templates/myhpom/upload/current_ad.html [hunk 1]
```diff
@@ -5,15 +5,66 @@
...
     <di>
         <dt class="advance-directive-widget__definition__key">active date:</dt>
-        <dd class="advance-directive-widget__definition__value">{{ advancedirective.valid_date|date:"SHORT_DATE_FORMAT" }}</dd>
+        <dd class="advance-directive-widget__definition__value">{{ request.user.advancedirective.valid_date|date:"SHORT_DATE_FORMAT" }}</dd>
     </di>
 </dl>
+
+<div class="row">
+    <div class="col">
+        <a class="advance-directive-widget__button--primary" href="{{ request.user.advancedirective.document.url }}" target="_blank" rel="noopener noreferrer">Download</a>
```
Comment (sharrisoncaktus) [inline myhpom/templates/myhpom/upload/current_ad.html]: This does not work when there is no document

**Review by sharrisoncaktus — COMMENTED** [review]
The main question I have is, how does this interact with the things that Dane has been working on? In particular, we decided not to do the upload itself in MH-77, but he was going to add it in MH-102. So you should probably review each other's pull requests now!

File: myhpom/views/upload.py [hunk 2]
```diff
@@ -77,3 +80,13 @@ def upload_sharing(request):
     return render(request, 'myhpom/upload/sharing.html', {
         'form': form,
     })
+
+
+@require_http_methods(['POST'])
+@login_required
```
Comment (dsummersl) [inline myhpom/views/upload.py]: Should this be only ajax? If so add `@require_ajax`

File: myhpom/views/upload.py [hunk 3]
```diff
@@ -42,10 +42,13 @@ def upload_requirements(request):
     else:
         directive = AdvanceDirective(user=request.user, share_with_ehs=False)
     MIN_YEAR = 1950
+
+    form = UploadRequirementsForm()
```
Comment (dsummersl) [inline myhpom/views/upload.py]: This appears to be replaced by the `else` branch a few lines down.

File: myhpom/views/upload.py [hunk 4]
```diff
@@ -77,3 +80,13 @@ def upload_sharing(request):
...
+
+
+@require_http_methods(['POST'])
+@login_required
+def upload_delete_ad(request):
+    if hasattr(request.user, 'advancedirective'):
+        request.user.advancedirective.delete()
+        return HttpResponseRedirect(reverse('myhpom:upload_index'))
+    else:
+        return HttpResponseRedirect(reverse('myhpom:upload_current_ad'))
```
Comment (dsummersl) [inline myhpom/views/upload.py:91]: Add a test case for this to test_upload_view.py

**Review by dsummersl — CHANGES_REQUESTED** [review]
Looks great - couple small things noted.

**Review by dsummersl — APPROVED** [review]
👍

**Pull request merged**